mirar/pipelines/winter/validator.py

```diff
@@ -0,0 +1,70 @@
+"""
+Module to reject images based on quality criteria
+"""
+import numpy as np
+
+from mirar.data import ImageBatch
+from mirar.processors.astrometry.validate import PoorAstrometryError, PoorFWHMError
+from mirar.processors.mask import TooManyMaskedPixelsError
+from mirar.processors.split import SUB_ID_KEY
+
+
+def masked_images_rejector(batch: ImageBatch) -> ImageBatch:
+    """
+    Rejects images with too many masked pixels
+    """
+    assert len(batch) == 1
```

Comment:
Why does the batch have to be len(1) ?

Reply:
So that one image with many masked pixels doesn't affect others.. I put this in to ensure that we don't accidentally forget to do the correct ImageBatching before using this function.

Reply:
OK, sounds reasonable